mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical shadow estimation for quantum circuits. Based on the paper"""
+
+from typing import (
+    Optional,
+    Callable,
+    List,
+    Dict,
+    Any,
+    Tuple,
+    Union,
+    Mapping,
+)
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+
+import mitiq
+from mitiq import MeasurementResult
+from mitiq.shadows.quantum_processing import random_pauli_measurement
+from mitiq.shadows.classical_postprocessing import (
+    shadow_state_reconstruction,
+    expectation_estimation_shadow,
+    get_pauli_fidelities,
+)
+
+
+def pauli_twirling_calibrate(
+    k_calibration: int = 1,
+    locality: Optional[int] = None,
+    zero_state_shadow_outcomes: Optional[Tuple[List[str], List[str]]] = None,
+    qubits: Optional[List[cirq.Qid]] = None,
+    executor: Optional[Callable[[cirq.Circuit], MeasurementResult]] = None,
+    num_total_measurements_calibration: Optional[int] = 20000,
+) -> Dict[str, complex]:
+    r"""
+    This function returns the dictionary of the median of means estimation
+    of Pauli fidelities: {:math:`\{'b':f_{b}\}_{b\in\{0,1\}^n}`}.
+
+    Args:
+        k_calibration: Number of groups of "median of means" used for
+            calibration.
```

Comment:
What does does the "median of means" mean here? I don't see any collection of means we could be taking a median of, so it's a little unclear.

Reply:
I don't think I could make it clear locally.  I have added references, which I believe users should know what they are calculating. The more clear reference has been given in the "What's the theory behind" part, as well as in the rshadows_tutorial.md notebook.